nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -0,0 +1,148 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.docker-containers;
+
+  dockerContainer =
+    { name, config, ... }: {
+
+      options = {
+
+        image = mkOption {
+          type = types.str;
+          description = "Docker image to run.";
+          example = "library/hello-world";
+        };
+
+        cmd = mkOption {
+          type =  with types; listOf str;
+          default = [];
+          description = "Commandline arguments to pass to the image's entrypoint.";
+        };
+
+        entrypoint = mkOption {
+          type = with types; nullOr str;
+          description = "Overwrite the default entrypoint of the image.";
+          default = null;
+        };
+
+        environment = mkOption {
+          type = with types; attrsOf str;
+          default = {};
+          description = "Environment variables to set for this container.";
+          example = {
+            DATABASE_HOST = "db.example.com";
+            DATABASE_PORT = "3306";
+          };
+        };
+
+        log-driver = mkOption {
+          type = types.str;
+          default = "none";
+          description = ''
+            Logging driver for the container.  The default of "none" means that
+            the container's logs will be handled as part of the systemd unit.
+            Setting this to "journald" will result in duplicate logging, but
+            the container's logs will be visible to the `docker logs` command.
+          '';
+        };
+
+        ports = mkOption {
+          type = with types; attrsOf str;
+          default = {};
+          description = "Network ports to forward from the host to this container.";
+          example = {
+            "8080" = "9000/tcp";
+          };
+        };
+
+        user = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = ''
+            Override the username or UID (and optionally groupname or GID) used
+            in the container.
+          '';
+          example = "nobody:nogroup";
+        };
+
+        volumes = mkOption {
+          # Note: these are "src:dst" lists so it's possible for `src`
+          # to refer to a /nix/store path, and for `dst` to include
+          # mount options.
```

Comment:
This should probably be included in the description

Reply:
done.